File: ios/EmpyrBridge.m
#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(EmpyrBridge, NSObject)
RCT_EXTERN_METHOD(initialize:(NSString)clientId)
RCT_EXTERN_METHOD(track: (nonnull NSNumber)offerId withTracker:(NSString)tracker)
@end

File: ios/EmpyrBridge.swift
// EmpyrBridge.swift
import Foundation

@objc(EmpyrBridge)
class EmpyrBridge: NSObject {
  
  @objc
  func initialize(_ clientId: NSString) {
    _ = EmpyrAPIClient.initialize(clientId:clientId as String)
  }

  @objc
  func track(_ offerId: NSNumber, withTracker tracker: NSString ) {
    if (tracker == "PROFILE_VIEW") {
      EmpyrAPIClient.mainInstance().track(offerId:offerId.intValue, tracker: Tracker.PROFILE_VIEW)
    } else if (tracker == "SEARCH_VIEW") {
        EmpyrAPIClient.mainInstance().track(offerId:offerId.intValue, tracker: Tracker.SEARCH_VIEW)
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
